### PR TITLE
[BUGS-2845] Added usage descriptions in help output

### PR DESCRIPTION
**Pull request by TeslaDethray**
### Disposition
This pull request:

- [x] Fixes a bug
- [ ] Adds a feature
- [ ] Breaks backwards compatibility
- [x] Has tests that cover changes

### Summary & Description
The descriptions are not being added to the usage output. When a description is provided, it is added to the examples property array but when the help docs are generated, the private `usages` property inside `Symfony\Component\Console\Command\Command` is used, which omits the descriptions.

### CODE REVIEW

**Comment by TeslaDethray** [comment]
Using `2.12.0`
Screen Shot 2019-10-21 at 17 09 13

Using `TeslaDethray:bugs-2845`
Screen Shot 2019-10-21 at 17 05 23

From [Pantheon\Terminus\Commands\Env\InfoCommand](https://github.com/pantheon-systems/terminus/blob/master/src/Commands/Env/InfoCommand.php#L40-L41)
```
    /**
     * Displays environment status and configuration.
     *
     * @authorize
     *
     * @command env:info
     *
     * @field-labels
     *     id: ID
     *     created: Created
     *     domain: Domain
     *     locked: Locked
     *     initialized: Initialized
     *     connection_mode: Connection Mode
     *     php_version: PHP Version
     *     drush_version: Drush Version
     * @return PropertyList
     *
     * @param string $site_env Site & environment in the format `site-name.env`
     *
     * @usage env:info <site>.<env>
     *    Displays status and configuration for <site>'s <env> environment.
     */
```

**Comment by greg-1-anderson** [comment]
This looks good, but we'll either have to bump the major version here and drop Symfony 2, or fix the Travis tests and prove that this works with Symfony 2.

I think that the code is the way it is because usage descriptions don't work in Symfony 2. Although I don't clearly remember the exact situation here, I'd be okay with a new major version that just drops old Symfony versions and old PHP versions. I'm doing similar things in other projects.

**Comment by TeslaDethray** [comment]
@greg-1-anderson Perhaps I'm using the `@usage` tag improperly. (I see that Symfony is not adding descriptions to their usage help output.) Is there a way to make use of `AnnotatedCommand`'s `examples` array? I see where it's being used by the `HelpDocumentBuilder`, but I'm not sure where this information surfaces.

**Comment by greg-1-anderson** [comment]
This information could not be displayed in Symfony 2, so AnnotatedCommand kept the info but never used it. Looks like this PR solves that problem, but I don't think it works on Symfony 2. I think that bumping the major version here is a good solution to that problem though.

File: composer.json [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
     },
     "require": {
-        "php": ">=5.4.5",
+        "php": ">=5.5.9",
         "consolidation/output-formatters": "^3.4",
         "psr/log": "^1",
-        "symfony/console": "^2.8|^3|^4",
-        "symfony/event-dispatcher": "^2.5|^3|^4",
-        "symfony/finder": "^2.5|^3|^4"
+        "symfony/console": "^3|^4",
```
Comment (TeslaDethray) [inline composer.json]: @greg-1-anderson I _think_ this is what you meant. LMK if I've misinterpreted.

**Comment by greg-1-anderson** [comment]
Yeah it looks good. The main thing is I need to make a 2.x branch before merging this; I've just been busy with DrupalCon coming up.

**Comment by greg-1-anderson** [comment]
The `main` branch now supports only Symfony 4 and Symfony 5, so this could be rebased against main, and we wouldn't need to worry about Symfony 2 behavior.

**Comment by TeslaDethray** [comment]
Closing due to age and irrelevance

**Pull request closed without merging**